### DISCUSSION TITLE
Running on Linux

### DISCUSSION THREAD

**Discussion by Rahix** [JoshParnell/ltheory-old#3] [Show and tell]
I got it running on Linux!  Archlinux to be precise.  FWIW, I had to take the following steps:

1. I needed to install `libpng12` package.
2. Apply the following change:
   ```diff
   diff --git a/CMakeLists.txt b/CMakeLists.txt
   index ada6f46..b4c7a34 100644
   --- a/CMakeLists.txt
   +++ b/CMakeLists.txt
   @@ -3,6 +3,10 @@ project(LTheory)
    
    find_package(OpenGL REQUIRED)
    
   +find_package(libpng12)
   +
   +set(CMAKE_POSITION_INDEPENDENT_CODE ON)
   +
    # force SFML to build as a static lib to minimize pain
    set(BUILD_SHARED_LIBS FALSE CACHE BOOL "" FORCE)
    add_subdirectory(ext/SFML)
   @@ -185,7 +189,7 @@ else()
      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2 -g")
    
      # Enable lots of warnings
   -  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall -Wextra -Werror")
   +  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall -Wextra")
      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-invalid-offsetof")
      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-const-variable")
      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
   ```
3. Upgrade the vendored SFML dependency to 2.5.1:
   ```bash
   cd ext/SFML
   git switch --detach 2.5.1
   ```
4. Build using
   ```bash
   mkdir build
   cd build
   cmake .. -G ninja
   ninja
   ```
5. Run it!  I found that instead of the "war" scenario that is suggested in the README, you can also run `ltheory` which spawns you next to a space-station with a highway (?) nearby.  Run it from the repository root like this:
   ```bash
   ./bin/launch ltheory
   ```

Still trying to figure out the keybinds, what I've found so far:

| Key | Binding |
| --- | --- |
| `W`, `A`, `S`, `D` | Basic directional thrust |
| `SPACE` | Turn on/off mouse rotation (ship rotates to follow cursor) |
| `Q`, `E` | Rotate camera left/right, but seemingly not rotating the ship |
| `Right Mouse` | Fire weapons |

According to some code, `Y`/`U` should be attach/detach for the highway (?) but that didn't work.  Maybe I missed something or it is just not working in this scenario.

Anyway, here is a picture for proof:

![limit-theory](https://user-images.githubusercontent.com/7566996/182280679-3015b621-8b7f-48e0-a894-0e6fc20b2909.png)

**Comment by paralin** [comment]
Applied your changes: https://github.com/JoshParnell/ltheory-old/pull/4

Gentoo needed `media-libs/libpng-compat:1.2`

LD_LIBRARY_PATH=$(pwd)/extbin/linux64 ./bin/launch ltheory

**Reply by paralin** [reply]
![image](https://user-images.githubusercontent.com/534887/183033623-56e9e5a5-c596-47c4-9f84-82108a482d46.png)

Running on Gentoo

**Comment by dw0xaa55** [comment]
I needed to add -fPIC to the FLAGS of some cpp files for ninja to be able to compile and link the final program.
Also I'm on Arch, by the way ;-)

![screenshot-04-01-2023_02-14-20](https://user-images.githubusercontent.com/70031528/210467278-ec4a0c8d-ea7a-4c5f-8820-1d62283ac2fa.png)
